tests/utils.py

```diff
@@ -55,10 +68,27 @@ def venv_python(venv_dir: Path) -> Path:
     return venv_dir / "bin" / "python"
 
 
+# ====================================================================
+# Parsing the requirements file
+# ====================================================================
+
+
 @cache
+def parse_versions_from_requirements() -> dict[str, Requirement]:
+    """Return the requested version for each package in the requirements file."""
+    reqs: dict[str, Requirement] = {}
+    for line in open("requirements-tests.txt"):
+        line = line.split("#")[0].strip()
+        if not line:
+            continue
+        req = Requirement(line)
+        reqs[req.name] = req
+    return reqs
+
+
 def get_mypy_req() -> str:
-    with open("requirements-tests.txt", encoding="UTF-8") as requirements_file:
-        return next(strip_comments(line) for line in requirements_file if "mypy" in line)
+    req = parse_versions_from_requirements()["mypy"]
+    return str(req.specifier)
```

Comment:
I think you just want this, don't you?

```suggestion
    return str(parse_versions_from_requirements()["mypy"])
```

Reply:
Yes. Made this error twice today. 🤦‍♂️